### PR TITLE
Add overview of using pubsub Simple* classes

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: apis/Google.Cloud.PubSub.V1/docs/index.md [hunk 1]
```diff
@@ -45,6 +45,15 @@ provide a general-purpose abstraction over raw the RPC API, providing
 features such as page streaming to make client code cleaner and
 simpler.
 
+[SimplePublisher](obj/api/Google.Cloud.PubSub.V1.SimplePublisher.yml) and [SimpleSubscriber](obj/api/Google.Cloud.PubSub.V1.SimpleSubscriber.yml) provide simpler APIs for message publishing and subscribing.
+
 # Sample code
 
+Using [PublisherClient](obj/api/Google.Cloud.PubSub.V1.PublisherClient.yml) and
```
Comment (jskeet) [inline apis/Google.Cloud.PubSub.V1/docs/index.md]: Possibly reorder these, to encourage the use of `Simple*`?

File: apis/Google.Cloud.PubSub.V1/docs/index.md [hunk 2]
```diff
@@ -45,6 +45,15 @@ provide a general-purpose abstraction over raw the RPC API, providing
 features such as page streaming to make client code cleaner and
 simpler.
 
+[SimplePublisher](obj/api/Google.Cloud.PubSub.V1.SimplePublisher.yml) and [SimpleSubscriber](obj/api/Google.Cloud.PubSub.V1.SimpleSubscriber.yml) provide simpler APIs for message publishing and subscribing.
```
Comment (jskeet) [inline apis/Google.Cloud.PubSub.V1/docs/index.md]: I think you can be bolder here, talking about efficiencies etc. (Or do that later...)

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs [hunk 3]
```diff
@@ -89,6 +89,56 @@ public void Overview()
...
+            publisher.CreateTopic(topicName);
+
+            // Subscribe to the topic.
+            SubscriberClient subscriber = await SubscriberClient.CreateAsync();
+            SubscriptionName subscriptionName = new SubscriptionName(projectId, subscriptionId);
+            subscriber.CreateSubscription(subscriptionName, topicName, pushConfig: null, ackDeadlineSeconds: 60);
+
+            // Publish a message to the topic using SimplePublisher.
+            SimplePublisher simplePublisher = await SimplePublisher.CreateAsync(topicName);
+            // The data can be any arbitrary ByteString.Here, we're using the string overload.
```
Comment (jskeet) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: Comment doesn't really make sense at the moment. Perhaps enumerate the overloads?

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs [hunk 4]
```diff
@@ -89,6 +89,56 @@ public void Overview()
...
+            string messageId = await simplePublisher.PublishAsync("Hello, Pubsub");
+            // SimplePublisher should be shutdown after use.
```
Comment (jskeet) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: Maybe explain what the TimeSpan is for?

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs [hunk 5]
```diff
@@ -89,6 +89,56 @@ public void Overview()
...
+            SimpleSubscriber simpleSubscriber = await SimpleSubscriber.CreateAsync(subscriptionName);
+            List<PubsubMessage> receivedMessages = new List<PubsubMessage>();
+            // Start the subscriber listening for messages.
+            await simpleSubscriber.StartAsync((msg, cancellationToken) =>
+            {
+                receivedMessages.Add(msg);
+                Console.WriteLine($"Received message {msg.MessageId} published at {msg.PublishTime.ToDateTime()}");
+                Console.WriteLine($"Text: '{msg.Data.ToStringUtf8()}'");
+                // Stop this subscriber after one message is received.
+                // This is non-blocking, and the returned Task may be awaited on.
```
Comment (jskeet) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: Remove the "on".

File: apis/Google.Cloud.PubSub.V1/docs/index.md (hunk 2 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.PubSub.V1/docs/index.md]: Done

File: apis/Google.Cloud.PubSub.V1/docs/index.md (hunk 1 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.PubSub.V1/docs/index.md]: Done.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs (hunk 3 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: Fixed.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs (hunk 4 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: Done.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs (hunk 5 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Snippets/SubscriberClientSnippets.cs]: Done.

**Pull request merged**